Develop add hwise testthat

Alb-Ramirez opened the pull request:
Hi both,

just some basic test to check that the function works.
@SaeedR1987 @ig-impact

berti83 committed:
Initialization of Testthat 3

berti83 committed:
Initialization of the testthat file for add_hwise

berti83 committed:
Add testthat tests

ig-impact commented:
Hi @Alb-Ramirez.

When I ran the tests I immediately got an error about an un-exported function `c_across` on lines 271 and 184.
Additionally, it might be good to lint the code being tested as well. I see for example this message:

     ├╴󰀪  Use `||` in conditional expressions. lintr (vector_logic_linter) [93, 32]

More importantly, I think we might need to work on the test cases a bit more. When running the tests I get these two failures:

──────────────────────────────────────────────────────────────────────────────
Failure (test-add_hwise.R:412:3): add_hwise assigns correct HWISE-12 severity 
categories
Expected `result$hwise12_severity_cat` to equal `c("No-to-marginal", "No-to-ma
rginal", "Moderate", "High")`.
Differences:
`actual`:   "No-to-marginal" "No-to-marginal" "Low"      "High"
`expected`: "No-to-marginal" "No-to-marginal" "Moderate" "High"

──────────────────────────────────────────────────────────────────────────────

══ Results ═══════════════════════════════════════════════════════════════════
── Failed tests ──────────────────────────────────────────────────────────────
Failure (test-add_hwise.R:412:3): add_hwise assigns correct HWISE-12 severity 
categories
Expected `result$hwise12_severity_cat` to equal `c("No-to-marginal", "No-to-ma
rginal", "Moderate", "High")`.
Differences:
`actual`:   "No-to-marginal" "No-to-marginal" "Low"      "High"
`expected`: "No-to-marginal" "No-to-marginal" "Moderate" "High"


[ FAIL 1 | WARN 0 | SKIP 0 | PASS 23 ]


Happy to hop in for a second review once these are addressed.

berti83 committed:
- Add the short-circuit operator on the add_hwise function to avoid raising warnings on lintr.
- Add the expressed call to dplyr when using c_across.
- Update the expect output in the test #13. It was worng the first time. The function works properly

Alb-Ramirez reviewed (commented):
I have updated the files as per Iyed's comments:
- Add the short-circuit operator on the add_hwise function to avoid raising warnings on lintr.
- Add the expressed call to dplyr when using c_across.
- Update the expect output in the test #13. It was worng the first time. The function works properly

ig-impact commented:
@QuentinVillotta for visibility
